libbeat/cmd/instance/beat.go

```diff
@@ -1083,13 +1083,16 @@ func initPaths(cfg *common.Config) error {
 	// the paths field. After we will unpack the complete configuration and keystore reference
 	// will be correctly replaced.
 	partialConfig := struct {
-		Path paths.Path `config:"path"`
+		Path   paths.Path `config:"path"`
+		Hostfs string     `config:"system.hostfs"`
 	}{}
 
 	if err := cfg.Unpack(&partialConfig); err != nil {
 		return fmt.Errorf("error extracting default paths: %+v", err)
 	}
 
+	partialConfig.Path.Hostfs = partialConfig.Hostfs
```

Comment:
Is this always safe? By introducing `Hostfs` to the `Path` struct we will allow users to overwrite the path in the configuration file already.

Reply:
Maybe I'm missing something, but is that any different from `-E` flags?

Reply:
Paths are configured twice I think. Once via `--path.X` on startup + `--system.hostfs`.

Then the config file is read and the path settings from the configuration file are applied again. The `-E` flags for path settings only become effective during the second phase. That is why the shim bash scripts just use `--path.X` CLI flags and we do not attempt to overwrite the defaults in the config.

The thing I was wondering: What if `path.hostfs` is configured, but `system.hostfs` is not? In that case you will set `path.hostfs` to an empty string here. Is this on purpose? Why not:

```
if partialConfig.Path.Hostfs == "" {
  partialConfig.Path.Hostfs = partialConfig.Hostfs
}
```

If the forced overwrite is on purpose a comment explaining the why would be helpful.

Reply:
Okay, so, technically speaking, there is no `path.hostfs`. There's only `system.hostfs`, and we just expand the main `Path` struct type with `hostfs`, hence why that's being overwritten. Ideally it would only be `path.hostfs`, but I'm mostly worried about breaking changes here, since I'm considering this to be primarily a bugfix so we can properly set `hostfs` in the cgroup self-monitoring.

Reply:
Ah, good point.